### PR TITLE
Remove Aqua references from notebooks

### PR DESCRIPTION
**Pull request by manoelmarques**
### Summary

Aqua imports replaced by QIskit Terra classes.
Chemistry, Finance, Machine Learning and Optimization were removed as all those were converted and migrated to the new application repositories

### CODE REVIEW

File: tutorials/algorithms/02_vqe_convergence.ipynb
Comment (woodsp-ibm) [inline tutorials/algorithms/02_vqe_convergence.ipynb:280]: Since we could not really run and show logging I had just copy/paste a couple of lines from an actual run into the notebook in the past. I think the lines are still valid as such but we should edit out the `.aqua` in them which would make them correct for the current code.

File: tutorials/algorithms/04_vqe_advanced.ipynb
Comment (woodsp-ibm) [inline tutorials/algorithms/04_vqe_advanced.ipynb:172]: > Here we see that result was arrived at much more quickly where `optimizer_evals` is 10 versus 72

the `10 versus 72` needs updating - it looks like `9 versus 65` now. I did look to see whether it was possible to substitute values derived from the code into the text in the past as I wondered if things might change, but that did not seem readily doable. Maybe since it has changed, and maybe will do again, perhaps we should print some message via the code to highlight the different rather than having this in a text section.

File: tutorials/algorithms/04_vqe_advanced.ipynb
Comment (woodsp-ibm) [inline tutorials/algorithms/04_vqe_advanced.ipynb:233]: In the text below this- again it includes values from the results that look like its needs updating

> The optimal value can be seen to be wrong too, i.e. -1.098 versus the correct -1.857.

The first value, the wrong one, looks now to be -1.074

File: tutorials/algorithms/09_textbook_algorithms.ipynb
Comment (woodsp-ibm) [inline tutorials/algorithms/09_textbook_algorithms.ipynb:9]: Should we rename the notebook itself too - say `09_factorizers` since this is all this notebook is about now

**Review by woodsp-ibm — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]
This LGTM, the only thing that sticks out is the use of BasicAer everywhere (since that's not really a change). We really shouldn't be using it in the tutorials because it's buggy and slow (it's only for places where aer isn't available) and we don't want to encourage it's use. But we can fix that in a follow up.

**Comment by manoelmarques** [comment]
I can still change all to use Aer in this PR if nobody has any objections. It should be a simple change.

**Comment by nonhermitian** [comment]
I think we need to leave this on hold for now, and wait until the next release to remove everything.  There is likely to be a non-trivial subset of users who either cannot or do not want to update to the new modules at this time, e.g. those in the middle of getting results / writing papers.  As such, these should all remain for now.

**Comment by woodsp-ibm** [comment]
@nonhermitian You are advocating keeping the former Aqua set of tutorials, that are being deleted by this until the next release? The one thing we need to address is algorithms and operators, since, unlike the applications areas of optimization etc we do not have a copy (new updated set) elsewhere. We could rename algorithms and operators that are here to something like algorithms_legacy and operators_legacy and the new ones based on the current code in Terra are in the original place, i.e. algorithms and operators. Tutorials index would just have to be adjusted to have both sets with the legacy ones having the deprecation banner.

**Comment by nonhermitian** [comment]
Indeed, I am saying we should keep the deprecated Aqua stuff around until the next major release.  The docs already say the relevant tutorials are deprecated, and one should look in the new modules for those items.

**Comment by mtreinish** [comment]
@nonhermitian I think we can consider this unblocked now. We're hosting stable versions of the documentation including tutorials for previous releases on qiskit.org/documentation, for example: https://qiskit.org/documentation/stable/0.24/index.html for 0.24 before the aqua deprecation. (they're also accessible/discoverable from the dropdown on the main documentation page). So if people are on old versions they can just refer to the older release's documentation.

**Comment by nonhermitian** [comment]
I think this would still be picked up in a point release would it not?

**Comment by nonhermitian** [comment]
So if we fine with a point removing the tutorials from 0.26 then ok by me.

File: tutorials/operators/02_gradients_framework.ipynb
Comment (mtreinish) [inline tutorials/operators/02_gradients_framework.ipynb:221]: In talking to @Cryoris he suggested replacing `~StateFn` with `StateFn(..., is_measurement=True)` to avoid the failure this is hitting. If we do that we should be able to uncomment this line

Comment (manoelmarques) [reply tutorials/operators/02_gradients_framework.ipynb]: I made the change and it fixed the problem.

**Comment by Cryoris** [comment]
I can reproduce the failures locally with 0.17 but I couldn't find a way to resolve it (like taking the conjugate of the coefficients) that works also with 0.18. I'd say just remove the coefficients and show how to compute the gradients with a fixed operator, i.e.
```
H = (2 * X) + Z
```

**Review by mtreinish — DISMISSED** [review]
LGTM, but I'll wait for @Cryoris, @stefan-woerner, or @woodsp-ibm  to confirm everything looks good here before merging since they have more domain expertise.

One thing I think we should do after this merges is to add a redirect from the old docs hosted tutorials to the application docs hosted versions. This should be pretty easy and something we can probably do by just adding entries to the redirect links like:
```
'tutorials/optimization/*": "https://qiskit.org/documentation/optimization/tutorials/$source"
```
to https://github.com/Qiskit/qiskit/blob/master/docs/conf.py#L82-L84

(see https://documatt.gitlab.io/sphinx-reredirects/usage.html for more details)

**Comment by peendebak** [comment]
@manoelmarques In the updated notebook `05_qaoa.ipynb` a different problem is solved than in the original notebook. QAOA is used for partitioning problem on a graph without taking the weights into account. The graph definition (and the corresponding plot) at the top if the notebook have weights in them. To avoid confusion I suggest to either i) update the notebook to solve the problem with weights, or ii) update the definition of the graph to have all edges weight 1.

**Comment by peendebak** [comment]
@manoelmarques Also from `05_qaoa.ipynb`. What is the goal of `get_graph_solution`? There is no explanation. Also if `x` is a solution, then `1-x` is a solution just as good as the objective function is invariant with respect to this operation.

File: tutorials/algorithms/03_vqe_simulation_with_noise.ipynb
Comment (Cryoris) [inline tutorials/algorithms/03_vqe_simulation_with_noise.ipynb:307]: Do we want to hide these deprecation warnings for the notebook? Or maybe update `measurement_error_mitigation`?

File: tutorials/algorithms/03_vqe_simulation_with_noise.ipynb
Comment (Cryoris) [inline tutorials/algorithms/03_vqe_simulation_with_noise.ipynb:232]: Is disabling parallelism still necessary? If yes, do we want to add a comment? 🙂

File: tutorials/algorithms/05_qaoa.ipynb
Comment (Cryoris) [inline tutorials/algorithms/05_qaoa.ipynb:285]: This solution doesn't match the others, do we know why?

File: tutorials/algorithms/06_grover.ipynb
Comment (Cryoris) [inline tutorials/algorithms/06_grover.ipynb]: This should print the oracle, otherwise we don't see that it uses 3 qubits 😄

**Review by Cryoris — COMMENTED** [review]
Overall looks good, only in the QAOA example it's strange that the NumPy result doesn't match the others. 

I left some other minor comments, but as these are general issues and not necessarily related to porting Aqua to Terra feel free to skip them 🙂

**Review by woodsp-ibm — APPROVED** [review]

File: tutorials/algorithms/03_vqe_simulation_with_noise.ipynb
Comment (manoelmarques) [reply tutorials/algorithms/03_vqe_simulation_with_noise.ipynb:307]: The deprecation msgs are eliminated in Qiskit/qiskit-terra#6503

File: tutorials/algorithms/03_vqe_simulation_with_noise.ipynb
Comment (manoelmarques) [reply tutorials/algorithms/03_vqe_simulation_with_noise.ipynb:232]: Removed statement

File: tutorials/algorithms/06_grover.ipynb
Comment (manoelmarques) [reply tutorials/algorithms/06_grover.ipynb:255]: Printing oracle now

**Review by woodsp-ibm — APPROVED** [review]

**Review by mtreinish — APPROVED** [review]

**Pull request merged**
